vistautils/parameters.py

```diff
@@ -1140,24 +1193,40 @@ def _interpolate(to_interpolate: Parameters, context: Parameters) -> Parameters:
         return Parameters.from_mapping(
             immutabledict(
                 (key, interpolation_mapping[key]) for key in to_interpolate._data.keys()
-            )
+            ),
+            namespace_prefix=to_interpolate.namespace_prefix,
         )
 
-    def _unify(self, old: Parameters, new: Parameters, namespace="") -> Parameters:
+    def _unify(
+        self,
+        old: Parameters,
+        new: Parameters,
+        *,
+        namespace_prefix: Sequence[str] = tuple(),
+    ) -> Parameters:
         # pylint:disable=protected-access
         ret = dict()
         for (key, old_val) in old._data.items():
             if key in new:
                 new_val = new._data[key]
                 if isinstance(old_val, Parameters) != isinstance(new_val, Parameters):
+                    if namespace_prefix:
+                        namespace_prefix_str = ".".join(namespace_prefix)
+                        param_str = f"{namespace_prefix_str}.{key}"
+                    else:
+                        param_str = key
+
                     raise IOError(
-                        "When unifying parameters, "
-                        + namespace
-                        + key
-                        + "is a parameter on one side and a namespace on the other"
+                        f"When unifying parameters, {param_str} is a parameter on one side and a "
+                        f"namespace "
```

Comment:
```suggestion
                        "namespace "
```
and join with the next line.